llvm/include/llvm/IR/DebugInfoMetadata.h

```diff
@@ -1048,6 +1088,39 @@ class DIDerivedType : public DIType {
     return DWARFAddressSpace;
   }
 
+  std::optional<PtrAuthData> getPtrAuthData() const;
+
+  /// \returns The PointerAuth key.
+  std::optional<unsigned> getPtrAuthKey() const {
```

Comment:
Are these helpers particularly useful? It seems like they change the usage code from this:
```
if (auto PtrAuthData = getPtrAuthData())
  do_something_with(PtrAuthData->Key());
```
to
```
if (auto Key = getPtrAuthKey())
  do_something_with(*Key);
```

Which doesn't seem like a huge win?

Reply:
These are just small shortcuts which do not give any value except what you described above. I personally have no strong preference, so accepting your suggestion - deleted the helpers. See f765247a3d49b3b726a7442d8d31c1cbd4782154